src/app/cours/cours.component.ts

```diff
@@ -12,45 +14,155 @@ import { Unit } from '../model/unit';
 })
 
 export class CoursComponent implements OnInit {
-  myClasses: Cours[] = [];
-  classForm: FormGroup;
-
-  constructor(private coursService: CoursService, private formBuilder: FormBuilder) {
-    this.classForm = this.formBuilder.group({
-      label: ['', Validators.required],
-      period: ['', Validators.required],
-      nbHour: ['',Validators.required],
-      teacher: ['', Validators.required],
-      detail: ['', Validators.required]
+  myClasses: Cours[] = []; classForm: FormGroup; profSelect: string;
+  profs: Professeur []; professor: Professeur [];
+
+  constructor(private coursService: CoursService, 
+              private profService: ProfService) {
+                
+    this.classForm = new FormGroup({
+      label: new FormControl('', Validators.required),
+      period: new FormControl('', Validators.required),
+      hours: new FormControl('',Validators.required),
+      professor: new FormControl('', Validators.required),
+      description: new FormControl('', Validators.required)
     });
   }
 
-  ngOnInit() {
-    this.myClasses = this.coursService.getCours();
+  ngOnInit(): void {
+   this.getCours();
+   this.getProfs();
+  }
+
+  getCours(): void{
+     this.coursService.getCours()
+    .subscribe(
+      (data)=> {
+        this.myClasses = data;},
+      (err)=> { console.error(err);}
```

Comment:
show en error using modal

Reply:
I don't understand, what you mean

Reply:
Oh yes, I understand now. Yes it's logic.